src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/TriviaDataFactory.CodeShapeAnalyzer.cs

```diff
@@ -106,6 +107,13 @@ private bool UseIndentation
 
             private static bool OnElastic(SyntaxTrivia trivia)
             {
+                // if this is structured trivia then we need to check for elastic trivia in any descendant
+                if (trivia.HasStructure)
+                {
+                    var structure = trivia.GetStructure()!;
+                    return structure.DescendantTrivia().Any(t => t.IsElastic());
```

Comment:
```suggestion
                    return structure.DescendantTrivia().Any(static t => t.IsElastic());
```

Reply:
Note: i don't love that i'm guessing that DescendantTrivia likely returns an IEnumerable, and we'll allocate for the enumerator as well. :-/

Consider writing a non-allocating walk here.